srv/salt/_modules/rgw.py

```diff
@@ -143,9 +127,93 @@ def _key(user, field, pathname):
 
 def access_key(user, pathname="/srv/salt/ceph/rgw/cache"):
     if not user:
-        raise ValueError("ERROR: no user specified") 
+        raise ValueError("ERROR: no user specified")
     return _key(user, 'access_key', pathname)
 
 def secret_key(user, pathname="/srv/salt/ceph/rgw/cache"):
     return _key(user, 'secret_key', pathname)
 
+def endpoints(cluster='ceph'):
+    result = []
+    search = "I@cluster:{}".format(cluster)
+    __opts__ = salt.config.client_config('/etc/salt/master')
+    pillar_util = salt.utils.master.MasterPillarUtil(search, "compound",
+                                                    use_cached_grains=True,
+                                                    grains_fallback=False,
+                                                    opts=__opts__)
+    cached = pillar_util.get_minion_pillar()
+    for minion in cached:
+        if 'rgw_endpoint' in cached[minion]:
+            match = re.search(r'http(s?)://(.+):?(\d*)', cached[minion]['rgw_endpoint'])
+            if match:
+                result.append({
+                    'host': match.group(2),
+                    'port': int(match.group(3)) if match.group(3) else 7480,
+                    'ssl': match.group(1) == 's',
+                    'url': cached[minion]['rgw_endpoint']
+                })
+            else:
+                result.append({
+                    'host': None,
+                    'port': None,
+                    'ssl': None,
+                    'url': cached[minion]['rgw_endpoint']
+                })
+            return result
+
+    port = '7480'  # civetweb default port
+    ssl = ''
+    admin_path = 'admin'
+    for rgw_conf_file_path in glob.glob("/srv/salt/ceph/configuration/files/ceph.conf.*"):
```

Comment:
we changed the logic for this is ui_rgw, not sure whether this would still be valid, what happens if there is an ceph.conf.rgw-ssl which we ship right now, wouldn't this just overwrite the value of host/port